documentation/modules/exploit/multi/misc/consul_service_exec.md

```diff
@@ -0,0 +1,116 @@
+## Vulnerable Application
+
+### Description
+
+This module exploits Hashicorp Consul's Services API to gain remote command execution on a Consul node.
+
+The exposure of the Services API depends on the `enable_script_checks` option. This option is opt-in for Consul nodes operators.
+
+#### References
+
+* Consul Services API - https://www.consul.io/api/agent/service.html
+* Inspiration from Garfield PoC - https://github.com/torque59/Garfield
+
+### Test setup
+
+The following bash script can be used to setup a testing environment with Docker:
+
+```
+#!/bin/sh
+
+echo "[+] Launching consul instances..."
+docker run -d --name=consul_bootstrap_server consul agent -server -client=172.17.0.1 -bootstrap -data-dir /tmp/consul
+sleep 2
+
+docker run -d --name=consul_server_1 consul agent -server -client=172.17.0.2 -data-dir /tmp/consul
+sleep 2
+docker exec -t consul_bootstrap_server consul join -http-addr="172.17.0.1:8500" 172.17.0.2
+
+docker run -d --name=consul_server_2 consul agent -server -client=172.17.0.3 -data-dir /tmp/consul
+sleep 2
+docker exec -t consul_bootstrap_server consul join -http-addr="172.17.0.1:8500" 172.17.0.3
+
+docker run -d -e 'CONSUL_LOCAL_CONFIG={"leave_on_terminate": true, "enable_script_checks":true}' consul agent -ui -client=172.17.0.4 -retry-join=172.17.0.1
+
+sleep 4
+echo "[+] Checking members..."
+docker exec -t consul_bootstrap_server consul members -http-addr="172.17.0.1:8500"
+```
```

Comment:
Please update this with the new test setup.

Reply:
And the output that corresponds to the new script.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/10443/commits/b9d4a469726507df4623cb7f42b939b9d37c237c